### PR TITLE
Additional Helm/Chart Configuration Updates

### PR DESCRIPTION
**Pull request by dealako**
* Updated/fixed Route and Image Stream apiVersion
* Added slave port to values and pod spec
* Added RSI:Apps docker registry support (push and pull)
* Added Image Pull Secrets (still need to refine/update to service account)
* Having trouble with volume claim template, commented out for now
* Added Jenking Location plugin configuration

JIRA: RE-1741

### CODE REVIEW

File: deployment/charts/jenkins-master/values.yaml [hunk 1]
```diff
@@ -4,38 +4,52 @@
...
+      port: "50000"
+      protocol: "tcp"
+      targetPort: "50000"
+      status:
+        probeDelaySeconds: 600
+        probeTimeoutSeconds: 60
   env:
     jenkinsArgs: "--prefix=/jenkins"
-    javaOpts: "-Dhudson.footerURL=\"http://www.rackspace.com\" -Djava.awt.headless=true -Djava.util.logging.config.file=\"/var/jenkins_home/logging.properties\""
+    javaOpts: "-Xmx2g -Xmx2g -Dhudson.footerURL=\"http://www.rackspace.com\" -Djava.awt.headless=true -Djava.util.logging.config.file=\"/var/jenkins_home/logging.properties\""
```
Comment (hughsaunders) [inline deployment/charts/jenkins-master/values.yaml]: one of these should probably be Xms

File: build-image.sh [hunk 2]
```diff
@@ -46,8 +46,8 @@ else
...
 
-log "Building Jenkins image with tag: ${_Y}${tag}${_W}..."
-docker build -t="${owner}/${project}:${version}" .
+log "Building Jenkins image with tag: ${_Y}${registry}/${namespace}/${project}:${version}${_W}..."
+docker build -t="${registry}/${namespace}/${project}:${version}" .
 
-log "Building Jenkins image with tag: ${_Y}latest${_W}..."
-docker build -t="${owner}/${project}:latest" .
+log "Building Jenkins image with tag: ${_Y}{registry}/${namespace}/${project}:latest${_W}..."
+docker build -t="${registry}/${namespace}/${project}:latest" .
```
Comment (hughsaunders) [inline build-image.sh]: We should probably only build once, store the image ID, and tag that rater than building twice.

File: deployment/charts/jenkins-master/values.yaml (hunk 1 above)
Comment (hughsaunders) [reply deployment/charts/jenkins-master/values.yaml]: Also why 2g when mem is 4g? To leave space for OS? I guess 3.5 would be ok?

Comment (dealako) [reply deployment/charts/jenkins-master/values.yaml]: yeah, I meant -Xmx and -Xms - usually the same value and historically, we need to set the JVM to half the total memory available (per Elasticsearch rules which were a good benchmark).

File: build-image.sh (hunk 2 above)
Comment (dealako) [reply build-image.sh]: yes! lazy

**Pull request merged**